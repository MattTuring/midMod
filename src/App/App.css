@import url('https://fonts.googleapis.com/css?family=Pacifico');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url('../assets/cafe-img.jpeg');
  width: 100%;
  height: 100vh;
  background-repeat: no-repeat;
  background-size: cover;
  text-align: center;
  grid-template-columns: 1fr 1fr 1fr;
}

.app-title {
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}

article{
  background-color: white;
  width: 25vw;
  border: 3px solid green;
  margin: 5px;
}

section {
  display: flex;
  justify-content: space-around;
}
