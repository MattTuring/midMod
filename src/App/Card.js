import React from 'react';

const Card = ({date, id, name, number, time, deleteReservation}) => {
  console.log(deleteReservation)
  return (
    <article>
    <p>{date}</p>
    <p>{name}</p>
    <p>{number}</p>
    <p>{time}</p>
    <button id={id} onClick={(event) => {deleteReservation(id); console.log(id)}}>Delete</button>
    </article>
  )
}

export default Card
