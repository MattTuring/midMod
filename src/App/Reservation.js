import React from 'react';

class Reservation extends React.Component {
  constructor(props) {
    console.log(props.newRes)
    super(props)
    this.state = {
        name: '',
        date: '',
        time: '',
        number: ''
    }
  }

sendFetch = () => {
  fetch('http://localhost:3001/api/v1/reservations', {
      method: 'POST',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify(this.state)
    })
  }


//New reservation: { id: 18939837, name: 'Leta', date: '12/3', time: '6:30', number: 2 }
  render() {
    return (
      <section>
        <input onChange={(event) => this.setState({name: event.target.value},() => {console.log(this.state)})} placeholder='Name'type='text'/>
        <input onChange={(event) => this.setState({date: event.target.value},() => {console.log(this.state)})} placeholder='Date ex. 12/3'type='text'/>
        <input onChange={(event) => this.setState({time: event.target.value},() => {console.log(this.state)})} placeholder='Time ex. 6:30'type='text'/>
        <input onChange={(event) => this.setState({number: parseInt(event.target.value)},() => {console.log(this.state)})} placeholder='Total People'type='number'/>
        <button onClick={() => {
          this.props.newRes(this.state)
          this.sendFetch()
        }}>Submit</button>
      </section>
    )
  }

}


export default Reservation
