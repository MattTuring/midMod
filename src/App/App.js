import React, { Component } from 'react';
import './App.css';
import Card from './Card'
import Reservation from './Reservation'

class App extends Component {
  constructor() {
    super()
    this.state = {
      cards: []
    }
  }
//{date, id, name, number, time}
  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
    .then(response => response.json())
    .then(data => {
      this.setState({cards:
      data.map(reservation => {
      return(
      <Card
        date={reservation.date}
        key={reservation.id}
        id={reservation.id}
        name={reservation.name}
        number={reservation.number}
        time={reservation.time}
        deleteReservation={this.deleteReservation}
      />)
    })
  })})
}

cardsUpdater = (reservation) => {
  let card = (<Card
    date={reservation.date}
    key={reservation.id}
    id={reservation.id}
    name={reservation.name}
    number={reservation.number}
    time={reservation.time}
    deleteReservation={this.deleteReservation}
  />)

  let res = this.state.cards.concat(card)

  this.setState({cards: res})
}

deleteReservation = (id) => {
  fetch(`http://localhost:3001/api/v1/reservations/${id}`,{
    method: 'DELETE'
  })

}

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
        <Reservation newRes={this.cardsUpdater}/>
        </div>
        <div className='resy-container'>
        {this.state.cards}
        </div>
      </div>
    )
  }
}

export default App;
