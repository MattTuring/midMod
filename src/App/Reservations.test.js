import React from 'react';
import ReactDOM from 'react-dom';
import Reservation from './Reservation';
import App from './App'
import { shallow } from 'enzyme';

describe('Reservation', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Reservation
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call function on click', () =>{
    const newRes = jest.fn();
  const wrapperApp = shallow(<App
    />)
  const wrapper = shallow(<Reservation
    newRes={newRes}
  />)

  wrapper.find('button').simulate('click')

   expect(newRes).toHaveBeenCalledWith(wrapper.state())
 })

 it('should set state on change', () =>{
const newRes = jest.fn();
 const wrapper = shallow(<Reservation
   newRes={newRes}
 />)

 wrapper.find('input').first().simulate('onChange')

  expect(wrapper.state()).toEqual({"date": "", "name": "", "number": "", "time": ""})
})

it('should call a function on click', () => {

  window.fetch = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        ok: true,
        json: () => Promise.resolve(mockResponse)
      })
    })
    const newRes = jest.fn();
  const wrapper = shallow(<Reservation
    newRes={newRes}
    />)

  wrapper.find('button').simulate('click')

  wrapper.instance().sendFetch(JSON.stringify({name:"Christie",date:"12/29",time:"7:00",number:12}))

  expect(window.fetch).toHaveBeenCalledWith("http://localhost:3001/api/v1/reservations", {"body": {"date": "", "name": "", "number": "", "time": ""}, "method": "POST"})
})

});
