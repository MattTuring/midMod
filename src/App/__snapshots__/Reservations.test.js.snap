// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation should match the snapshot 1`] = `
<section>
  <input
    onChange={[Function]}
    placeholder="Name"
    type="text"
  />
  <input
    onChange={[Function]}
    placeholder="Date ex. 12/3"
    type="text"
  />
  <input
    onChange={[Function]}
    placeholder="Time ex. 6:30"
    type="text"
  />
  <input
    onChange={[Function]}
    placeholder="Total People"
    type="number"
  />
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</section>
`;
